### DISCUSSION TITLE
How do you do email?

### DISCUSSION THREAD

**Discussion by chilltrout** [docmost/docmost#540] [Help / Questions]
Can someone show me an example of how the email is formatted?
I thought I had it right I'm not getting any errors in the logs but no emails get sent. 

> services:
>   docmost:
>     image: docmost/docmost:latest
>     depends_on:
>       - db
>       - redis
>     environment:
>       APP_URL: 'http://localhost:3000'
>       APP_SECRET: '********'
>       MAIL_DRIVER: 'postmark'
>       POSTMARK_TOKEN: '**************************************'
>       MAIL_FROM_ADDRESS:  'whatever@blah.com'
>       MAIL_FROM_NAME: 'invites'
>       DATABASE_URL: 'postgresql://docmost:********@db:5432/docmost?schema=public'
>       REDIS_URL: 'redis://redis:6379'
>     ports:
>       - "3000:3000"
>     restart: unless-stopped
>     volumes:
>       - docmost:/app/data/storage

should it go in the compose file or do I need to put it in a .env file?

**Comment by Philipinho** [comment]
Do you have an account with Postmark?

**Comment by matevz-ap** [comment]
``` docker
environment:
      APP_SECRET: $SERVICE_BASE64_APPKEY
      APP_URL: $SERVICE_FQDN_DOCMOST_3000
      DATABASE_URL: 'postgresql://$SERVICE_USER_POSTGRES:$SERVICE_PASSWORD_POSTGRES@postgresql/docmost?schema=public'
      REDIS_URL: 'redis://redis:6379'
      MAIL_DRIVER: $MAIL_DRIVER
      POSTMARK_TOKEN: $POSTMARK_TOKEN
      MAIL_FROM_ADDRESS: $MAIL_FROM_ADDRESS
      MAIL_FROM_NAME: $MAIL_FROM_NAME
```
If your environment is correctly configured, you should see errors in your docmost image logs. If not, check that the image has all 4 variables present. 

If you are by any chance using Coolify I had an issue where environment variables were not passed to docker.

**Reply by chilltrout** [reply]
no Im just using SMTP and then I tried postmark but neither work and no logs 
I have all 4 variables present it just wont send emails

**Reply by Philipinho** [reply]
@chilltrout please paste your server logs after attempting to send a mail.
